### PR TITLE
Kafka Connect Mongo Depositor

### PR DESCRIPTION
**Pull request by Michael7371**
This PR includes the following:

- Kafka Connect service with an initial configuration for MongoDB. It can also connect to S3, GCS, etc ([Brief Description of Kafka Connect](https://www.confluent.io/blog/kafka-connect-tutorial/))
- Example config file includes single message transforms to add a MongoDB recognized datetime field at the root to use for TTL indexes or other queries. Also adds support for converting ID fields to be used as the document ID when it is stored in Mongo.

### CODE REVIEW

File: Dockerfile
Comment (dan-du-car) [inline Dockerfile]: The maintainer instruction is deprecated, please remove it or replace it with LABEL. details:https://docs.docker.com/engine/reference/builder/#maintainer-deprecated ?

File: .gitignore
Comment (dan-du-car) [inline .gitignore]: Do not see this connect_start.sh.

File: src/main/resources/log4j.properties
Comment (dan-du-car) [inline src/main/resources/log4j.properties:9]: What is the difference between the log4j WARN logger and DEBUG appender?

File: mongo-connector/Dockerfile
Comment (dan-du-car) [inline mongo-connector/Dockerfile]: Why do we choose the name sample_connect_start.sh rather than connect_start.sh? Also if sample_connect_start.sh is the preferred name outside container. Maybe update the copy command: 
```
COPY sample_connect_start.sh /scripts/connect_start.sh
```

File: .gitignore
Comment (Michael7371) [reply .gitignore]: I am planning on re-working this, it will be updated after I can get the changes merged into the CDOT fork.

File: mongo-connector/Dockerfile
Comment (Michael7371) [reply mongo-connector/Dockerfile]: The original idea is that the sample_connect_start.sh would be a starting point to create a connect_start.sh. This seems slightly confusing after further thought, and instead I am going to push some changes that allow for attaching different connect_start.sh files via environmental variables rather than embedding it in the image.

File: docker-compose-mongo.yml
Comment (dan-du-car) [inline docker-compose-mongo.yml:53]: Is this MONGO_IP always the same as the DOCKER_HOST_IP? Maybe use one variable if yes. I got this issue testing the connector in local:
image

image

It cannot connect to Kafka message broker.

Comment (Michael7371) [reply docker-compose-mongo.yml:53]: In some cases, we do want to be able to vary the mongo_ip for more production deployments where a true replica-set mongodB deployment is desired.
I was able to reproduce the error you are seeing when I attempted to create connectors that didn't have a kafka topic already made for them. Can you try updating the KAFKA_CREATE_TOPICS variable (line 16 in docker-compose.yml) to the following:

KAFKA_CREATE_TOPICS: "topic.OdeSpatJson:1:1,topic.OdeBsmJson:1:1,topic.OdeTimJson:1:1,topic.OdePsmJson:1:1,topic.OdeRawEncodedPsmJson,topic.OdeRawEncodedBSMJson:1:1,topic.OdeRawEncodedSPATJson:1:1,topic.OdeRawEncodedTIMJson:1:1,topic.OdeRawEncodedMAPJson:1:1,topic.OdeMapJson:1:1,topic.OdeRawEncodedPSMJson:1:1,topic.OdePsmJson:1:1"

Comment (Michael7371) [reply docker-compose-mongo.yml:53]: I have added this fix to a current PR in the CDOT fork.

Comment (dan-du-car) [reply docker-compose-mongo.yml:53]: Updated the create topics list. Still in the loop waiting for 200. Also tried curl command in host machine, I got this result:
```
dev@dev:~$ curl http://connect:8083/connectors
curl: (6) Could not resolve host: connect
```
This is the env settings:
```
DOCKER_HOST_IP=localhost
# MONGODB Variables
MONGO_IP=${DOCKER_HOST_IP}
MONGO_PORT=27017

```

Comment (Michael7371) [reply docker-compose-mongo.yml:53]: Oh I see, localhost will not work. This will cause the kafka connect image to try to look at its own 9092 port for the kafka broker instead of through your host's network. Try runing 'ifconfig' in a wsl terminal and use the docker0 network or eth0 network IP.

Comment (dan-du-car) [reply docker-compose-mongo.yml:53]: I tried the docker0 IP, and it connected to Kafka message broker. However, the PUT request from the .sh script return 400 response multiple times within duration of 2 mins after containers are launched. 
image

Comment (Michael7371) [reply docker-compose-mongo.yml:53]: Interesting, can you share your current .env file and run the following commands for me in a WSL terminal and let me know the results:
sudo apt-get update
sudo apt-get install kafkacat
kafkacat -b localhost:9092 -G test_consumer_group topic.OdeSpatJson -f '\nTopic: %t\nKey: %k\nTimestamp: %T\nValue: %s\n' -o beginning

install mongosh in wsl/windows: https://www.mongodb.com/docs/mongodb-shell/install/
mongosh "mongodb://localhost:27017/?DirectConnection=true"

Also, the changes mentioned previously have now been merged into this PR so if you pull, your .env must be updated with the sample.env changes.

Comment (dan-du-car) [reply docker-compose-mongo.yml:53]: I am using ubuntu 22.04 Jammy distribution not windows.  Installed KafkaCat.
Pulled the latest dev branch and ran the command to build images and launch containers:  
` docker-compose -f docker-compose-mongo.yml up`
It seems up, then bring it down and up again. I got below issues:
image

Comment (Michael7371) [reply docker-compose-mongo.yml:53]: Okay, Ubuntu should work as well. The Mongodb password is required to set in your .env file. The docker0 network looks like the correct network to use.

Comment (Michael7371) [reply docker-compose-mongo.yml:53]: Also now that the kafka connect image has changed please run the following command to bring the containers down and back up again:

docker-compose -f docker-compose-mongo.yml down -v
docker-compose -f docker-compose-mongo.yml up --build --force-recreate --no-deps -d

Comment (dan-du-car) [reply docker-compose-mongo.yml:53]: image

Comment (Michael7371) [reply docker-compose-mongo.yml:53]: @dan-du-car We have just pushed some updates that should fix that exit code and other related issues.

**Review by dan-du-car — APPROVED** [review]

File: docker-compose-mongo.yml
Comment (dan-du-car) [reply docker-compose-mongo.yml:53]: Look good. Thank you!

**Pull request merged**
